src/main.c

```diff
@@ -101,6 +100,112 @@ static void init_sdl(void) {
 	log_info("Using SDL %u.%u.%u", v.major, v.minor, v.patch);
 }
 
+static char* get_default_res_path(void) {
+	char *res;
+
+#ifdef RELATIVE
+	res = SDL_GetBasePath();
+	strappend(&res, "data/");
+#else
+	res = strdup(STATIC_RESOURCE_PREFIX);
+#endif
+
+	return res;
+}
+
+static void get_core_paths(char **res, char **storage) {
+	if((*res = getenv("TAISEI_RES_PATH")) && **res) {
+		*res = strdup(*res);
+	} else {
+		*res = get_default_res_path();
+	}
+
+	if((*storage = getenv("TAISEI_STORAGE_PATH")) && **storage) {
+		*storage = strdup(*storage);
+	} else {
+		*storage = SDL_GetPrefPath("", "taisei");
+	}
+}
+
+static bool filter_zip_ext(const char *str) {
```

Comment:
Somehow I haven't even thought about other extension checking. It's not critical since we only use it for our resources so far (all of which are lowercase), but you're right that'd be more correct.

Maybe we should add a case-sensitivity parameter to the str{starts,ends}* functions instead, to not duplicate code. strendswith_any is also used for extension testing.